AudibleApi/Api.Library.cs

```diff
@@ -319,22 +319,24 @@ public async IAsyncEnumerable<Item[]> GetLibraryItemsPagesAsync(LibraryOptions l
 			int totalPages = totalItems / numItemsPerRequest;
 			if (totalPages * numItemsPerRequest < totalItems) totalPages++;
 
-			//Spin up as many concurrent downloads as we can/need
-			while (semaphore.CurrentCount > 0 && page < totalPages)
-				spinupPageRequest();
+			//Spin up as many concurrent downloads as we can/need. Minimum 1.
+			do
+				await spinupPageRequest();
+			while (semaphore.CurrentCount > 0 && page < totalPages);
 
 			while (pageDlTasks.Count > 0)
 			{
 				var completed = await Task.WhenAny(pageDlTasks);
 				pageDlTasks.Remove(completed);
 
-				//Request a new page.
-				if (page < totalPages) spinupPageRequest();
+				//Request new page(s)
+				while (semaphore.CurrentCount > 0 && page < totalPages)
+					await spinupPageRequest();
 
 				yield return completed.Result.Items;
 			}
 
-			async void spinupPageRequest()
+			async Task spinupPageRequest()
```

Comment:
I highly recommend naming this `spinupPageRequestAsync`. I think seeing that might have avoided this bug.

Reply:
Good idea. Done.